Issue #3236 Update attribute requirement levels in hardware metrics semconv (#3258)
open-telemetry/opentelemetry-specification: specification/metrics/semantic_conventions/hardware-metrics.md

All metrics in `hw.` instruments should be attached to a [Host Resource](../../resource/semantic_conventions/host.md)
and therefore inherit its attributes, like `host.id` and `host.name`.

> Additionally, all metrics in `hw.` instruments should have the below attributes:

Reviewer: ```suggestion
Additionally, all metrics in `hw.` instruments have the following attributes:
```

Author: done, thx!

**Description:** A memory module in a computer system.

| Name             | Description                                                                   | Units    | Instrument Type ([*](README.md#instrument-types)) | Value Type | Attribute Key            | Attribute Values                                |
| ---------------- | ----------------------------------------------------------------------------- | -------- | ------------------------------------------------- | ---------- | ------------------------ | ----------------------------------------------- |

> | `hw.errors`      | Number of errors encountered on this memory module                            | {errors} | Counter                                           | Int64      | `hw.type` (**Required**) | `memory`                                        |

Reviewer 2: See #3298.
```suggestion
| `hw.errors`      | Number of errors encountered on this memory module                            | {error} | Counter                                           | Int64      | `hw.type` (**Required**) | `memory`                                        |
```

| Name                                     | Description                                                                                 | Units    | Instrument Type ([*](README.md#instrument-types)) | Value Type | Attribute Key                   | Attribute Values                                |
| ---------------------------------------- | ------------------------------------------------------------------------------------------- | -------- | ------------------------------------------------- | ---------- | ------------------------------- | ----------------------------------------------- |

> | `hw.errors`                              | Number of errors encountered on this disk                                                   | {errors} | Counter                                           | Int64      | `hw.type` (**Required**)        | `physical_disk`                                 |

Reviewer 2: See #3298.
```suggestion
| `hw.errors`                              | Number of errors encountered on this disk                                                   | {error} | Counter                                           | Int64      | `hw.type` (**Required**)        | `physical_disk`                                 |
```

**Description:** A tape drive in a computer or in a tape library (excluding
virtual tape libraries)

| Name                       | Description                                                                   | Units        | Instrument Type ([*](README.md#instrument-types)) | Value Type | Attribute Key            | Attribute Values                             |
| -------------------------- | ----------------------------------------------------------------------------- | ------------ | ------------------------------------------------- | ---------- | ------------------------ | -------------------------------------------- |
| `hw.errors`                | Number of errors encountered by the tape drive                                | {errors}     | Counter                                           | Int64      | `hw.type`                | `tape_drive`                                 |

> | `hw.tape_drive.operations` | Operations performed by the tape drive                                        | {operations} | Counter                                           | Int64      | `type` (Recommended)     | `mount`, `unmount`, `clean`                  |

Reviewer 2: See #3298.
```suggestion
| `hw.tape_drive.operations` | Operations performed by the tape drive                                        | {operation} | Counter                                           | Int64      | `type` (Recommended)     | `mount`, `unmount`, `clean`                  |
```
